nixos/tests/foundationdb.nix

```diff
@@ -0,0 +1,35 @@
+import ./make-test-python.nix (
+  {
+    pkgs,
+    lib,
+    package ? pkgs.foundationdb,
+    ...
+  }:
+  {
+    name = "foundationdb";
+    meta.maintainers = with lib.maintainers; [ siriobalmelli ];
+
+    nodes = {
+      server =
+        { _, ... }:
+        {
+          services.foundationdb = {
+            inherit package;
+            enable = true;
+            serverProcesses = 1;
+            traceFormat = "json";
+          };
+
+          # fail test if unable to start the first time
+          systemd.services.foundationdb.serviceConfig.Restart = lib.mkForce "no";
+        };
+    };
+
+    testScript =
+      { nodes, ... }:
+      ''
+        server.wait_for_unit("foundationdb.service")
+        server.succeed("fdbcli --exec status")
```

Comment:
Should we add more test cases like adding and reading stuff to the database?

Reply:
I agree, but for now "starting and initializing the db successfully without coredump" is the bar to clear.